pints/_log_likelihoods.py

```diff
@@ -861,6 +862,302 @@ def evaluateS1(self, x):
+            \right) \right) \\
+            & + \sum_{\substack{x_{ij} = \text{upper}_{i}, \\1 \leq i
+            \leq n_t, \\ 1 \leq j \leq n_o}}  \log\left(1 -
+            \Phi \left(\frac{\text{upper}_{i} - f_j(t_i, \theta)}{\sigma_{i}}
+            \right) \right)
+        \end{align}
+
+    Extends :class:`ProblemLogLikelihood`.
+
+    Parameters
+    ----------
+    problem
+        A :class:`SingleOutputProblem` or :class:`MultiOutputProblem`.
+    lower
+        The lower limit for censoring.
+    upper
+        The upper limit for censoring.
+    verbose
+        When True, the input data and the values that are censored are printed.
+
+    References
+    ----------
+    .. [3] Beal, S. L. (2001). Ways to fit a PK model with some data below the
+           quantification limit. Journal of pharmacokinetics and
+           pharmacodynamics 28, pp. 481-504.
+
+    """
+
+    def __init__(self, problem, lower=None, upper=None, verbose=True):
+        super(CensoredGaussianLogLikelihood, self).__init__(problem)
+
+        # Get number of times, number of outputs
+        self._nt = len(self._times)
+        self._no = problem.n_outputs()
+
+        # Add parameters to problem
+        self._n_parameters = problem.n_parameters() + self._no
+
+        # Convert the lower and upper limits to the correct type
+        a = self._convert_type(lower, limit_type="lower")
+        b = self._convert_type(upper, limit_type="upper")
+
+        if len(a) != self._no:
+            raise ValueError(
+                'Lower limit must be a ' +
+                ' scalar or a vector of length n_outputs.')
+
+        if len(b) != self._no:
+            raise ValueError(
+                'Upper limit must be a ' +
+                ' scalar or a vector of length n_outputs.')
+
+        diff = b - a
+        if np.any(diff <= 0):
+            raise ValueError('Upper limit ' +
+                             'must exceed lower limit.')
+        self._a = a
+        self._b = b
+
+        # Define the condition for whether a point is not censored
+        self._condition = (self._a < self._values
+                           ) & (self._values < self._b)
+
+        # Number of points that aren't censored (for each observation
+        # in the multioutput case)
+        self._n_not_censored = np.sum(self._condition, axis=0)
+
+    def _convert_type(self, limit, limit_type="lower"):
+
+        # If no lower or upper limit is supplied set it equal to
+        # -/+ infinity
+        if limit is None:
+            if limit_type == "lower":
+                limit = - np.inf
+            elif limit_type == "upper":
+                limit = np.inf
+
+        # Convert the limit to an object of the correct type
+        if limit is not None:
+            if np.isscalar(limit):
+                limit = np.ones(self._no) * float(limit)
+            else:
+                limit = pints.vector(limit)
+
+        return limit
+
+    def print_censored_values(self):
```

Comment:
Cosmetic comment:

Methods are typically ordered alphabetically with underscores coming first. So this method should come after the __call__ method.

Reply:
And the evaluateS1 method.